### PR TITLE
Demo changes

### PR DESCRIPTION
**Pull request by shubhwale**
1. Login/Logout button and Welcome user text moved to right
2. Cart items are counted dynamically and badge shown for same.
3, Cart service modified and it uses localstorage instead of basicsubject class object

### CODE REVIEW

File: src/app/shared/book.service.ts [hunk 1]
```diff
@@ -3,68 +3,100 @@ import { HttpClient, HttpErrorResponse } from "@angular/common/http";
...
-        console.error('An error occurred:', error.error.message);
-      } else {
-        // The backend returned an unsuccessful response code.
-        // The response body may contain clues as to what went wrong,
-        console.error(`Backend returned code ${error.status}, ` + `body was: ${error.error}`);
+    str += month + "-" + day;
+    return str;
+  }
+
+  addToCartService(book: Book) {
```
Comment (ketannaik) [inline src/app/shared/book.service.ts:61]: Add to cart should be in Cart service instead of Book Service

File: src/app/shared/book.service.ts [hunk 2]
```diff
@@ -3,68 +3,100 @@ import { HttpClient, HttpErrorResponse } from "@angular/common/http";
...
+        }
+        else {
+          continue loop;
+        }
       }
-      // return an observable with a user-facing error message
-      return throwError('Something bad happened. Please try again later.');
     }
+    if (status == "Success") {
+      this.toastr.success(book.title + " Added", status);
```
Comment (ketannaik) [inline src/app/shared/book.service.ts:83]: Service code should not have toastr functionality in it..
Toastr is the messages which is shown from the component and not service.
Service should send either success or failure or any messages

File: src/app/books/book-details/book-details.component.ts [hunk 3]
```diff
@@ -36,9 +36,29 @@ export class BookDetailsComponent implements OnInit {
   }
 
   addToCart() {
-    var status : string = this.cart.sendProduct(this.book);
+    var status : string="Duplicate";
```
Comment (ketannaik) [inline src/app/books/book-details/book-details.component.ts:39]: This add to cart should be replaced by Cart Service function

File: src/app/header/header.component.ts [hunk 4]
```diff
@@ -1,16 +1,67 @@
...
+
+  ngAfterContentChecked(): void {
+    this.updateCartCount();
   }
 
+  updateCartCount(): void {
+    const localStoredCartItems: Cart[] = JSON.parse(localStorage.getItem("cartItems"));
+    if (localStoredCartItems.length > 0) {
+      this.cartItemsCountIcon.nativeElement.classList.remove('mat-badge-hidden');
+      this.count = localStoredCartItems.length;
```
Comment (ketannaik) [inline src/app/header/header.component.ts:37]: I am guessing this.count is the count of the cart items right??

Please follow proper naming conventions

File: src/app/app.component.ts [hunk 5]
```diff
@@ -7,4 +7,10 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'token-based-app';
```
[the remark this answers is not in the document]
Comment (shubhwale) [reply src/app/app.component.ts]: yeah changed it but committed in the wrong branch. Updated here as well. Sorry

File: src/app/auth/auth.interceptor.ts [hunk 6]
```diff
@@ -3,14 +3,15 @@ import { Injectable } from '@angular/core';
...
 
 @Injectable()
 export class AuthInterceptor implements HttpInterceptor {
-    constructor(private router : Router) {
+    constructor(private router : Router, private toastr: ToastrService) {
         
     }
     intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-        if(localStorage.getItem('accessToken')!=null) {
+        if(localStorage.getItem('accessToken')!=null && localStorage.getItem("refreshToken")) {
```
[the remark this answers is not in the document]
Comment (shubhwale) [reply src/app/auth/auth.interceptor.ts]: committed in wrong branch by mistake. Updated here

File: src/app/shared/book.service.ts (hunk 1 above)
Comment (shubhwale) [reply src/app/shared/book.service.ts]: Yes done

File: src/app/shared/book.service.ts (hunk 2 above)
Comment (shubhwale) [reply src/app/shared/book.service.ts]: done

File: src/app/books/book-details/book-details.component.ts (hunk 3 above)
Comment (shubhwale) [reply src/app/books/book-details/book-details.component.ts]: done

File: src/app/header/header.component.ts (hunk 4 above)
Comment (shubhwale) [reply src/app/header/header.component.ts]: yes that change was committed in another branch my mistake my bad will update here as well

**Pull request merged**